prov/rxm/src/rxm.h

```diff
@@ -639,6 +639,13 @@ RXM_EP_DEFINE_PROTO_LIMIT(inject)
 RXM_EP_DEFINE_PROTO_LIMIT(eager)
 RXM_EP_DEFINE_PROTO_LIMIT(sar)
 
+static inline
+size_t rxm_ep_get_genuine_inject_buf_limit(struct rxm_ep *rxm_ep)
+{
+	return (rxm_ep_get_inject_buf_limit(rxm_ep) ?
+		(rxm_ep_get_inject_buf_limit(rxm_ep) - 1) : 0);
+}
```

Comment:
This patch looks like it's adding more complex code that will be difficult for a future reader to understand what's going on.  And I don't understand the benefit of setting the limit to the wrong value, such that we need to call this function in several places just to adjust it back.

At the very least, if we're going to store something other than what someone will assume is the inject size in the inject_size field, it needs to be renamed.  Maybe inject_threshold?  Second, if we can't just have checks that work directly against the threshold, then I would consider storing 2 fields, so we don't have to add branches and subtraction operations into the code paths where the inject size needs to be checked.

Although my preference is to find another solution entirely.